---
layout: page
title: Configuring a source
---
<h3>General options when configuring a tier</h3>

<h5>name (required)</h5>

The name of the track, as shown in the tier-identification tab.

<h5>desc</h5>

A longer description of the content, shown when clicking on the tier-identification tab.  Currently only
plain text supported -- should probably allow a subset of HTML as well (TBD).

<h5>disabled</h5>

If <code>true</code>, the source is disabled by default (can be manually re-enabled from the
track selector).

<h5>stylesheet_uri</h5>

URI of a <a href='http://www.biodas.org/documents/spec-1.6.html#stylesheet'>DAS Stylesheet</a>
to use when displaying this data.  For DAS sources, this overrides the source's default stylesheet.
For other sources, this allows richer style configuration.  More information about Dalliance's implementation
of DAS stylesheets can be found <a href='stylesheets.html'>here</a>.

<h5>style</h5>

Array of <a href='stylesheets.html#json'>JSON-encoded style rules</a>.

<h5>mapping</h5>

Remap the data for this source using a mapping specified in the global <code>chains</code> configuration.  See <a href="config.html#liftOver">the section on coordinate system mappings</a>
for more details.

<h5>collapseSuperGroups</h5>

Hint to allow more "gene-like" rendering for some data sources.  Will probably be deprecated once
we find a better way of determining which sources provide gene annotations.

<h5>pinned</h5>

If <code>true</code>, the tier will be placed in the non-scrolling "pinned" section at the top of
the browser.

<h5>featureInfoPlugin</h5>

Configure a <a href='plugins.html'>feature-info plugin</a> which just applies to this tier.  <i>e.g.</i>:

{%highlight javascript%}
{name:                'BBD test',
 bwgURI:              'http://www.biodalliance.org/datasets/ensGene.bb',
 collapseSuperGroups: true
 featureInfoPlugin:   function(f, info) {
                          info.add('Testing', 'This is a test');
                      };
}
{%endhighlight%}


<h5>noSourceFeatureInfo</h5>

Disable any feature info plugins that would normally be added by the data source for this tier.

<h3>Configuring a DAS source</h3>

<h5>uri (required)</h5>

Base URI for the DAS server.  Used to construct URIs for requesting features, sequence, and
(if not over-ridden) stylesheets.  <i>e.g.</i>

{%highlight javascript%}

{name:                 'MeDIP raw',
 uri:                  'http://www.derkholm.net:8080/das/medipseq_reads'}
{%endhighlight%}

<h5>tier_type</h5>

Setting this to the string <code>'sequence'</code> indicates that this should be a DAS sequence
tier rather than a DAS features tier.

<h5>maxBins</h5>

Enables/disables the <code>maxbins</code> DAS extension for server-side downsampling
of quantitative data.  Default is <code>true</code>.

<h3>Configuring a binary sequence source</h3>

<h5>twoBitURI (required)</h5>

URI for the .2bit sequence file.  Alternatively, <code>twoBitBlob</code> can be used to configure
a local file as the sequence source.

<h5>tier_type</h5>

Setting this to the string <code>'sequence'</code> indicates that this is a sequence tier.

<h3>Configuring a bigwig or bigbed source</h3>

BigWig and BigBed files are configured in the same manner -- Dalliance automatically distinguished between
the two when loading.  big* sources are recognized by the presence of a <code>bwgURI</code> option in
their configuration.

<h5>bwgURI (required)</h5>

The URI for the big* file.  Despite the name, this can be used with both BigWig and BigBed files.

<h5>forceReduction</h5>

Takes an integer parameter.  Forces one of the 'reduced' forms of the data to be used.  0=first reduction, 1=second reduction, etc.
Alternatively, a value less than zero can be used to ensure that reduced data is <strong>never</strong> used.

<h5><a id='quantLeapThreshold'>quantLeapThreshold</a> (currently bigwig only)</h5>

Enables leap-by-threshold functionality.  Takes a numerical parameter which is the default threshold.

<h3>Configuring a BAM source</h3>

Dalliance can directly view BAM files so long as they're sorted and indexed.  BAM source configurations are recognized by
the presence of a bamURI option.

<h5>bamURI (required)</h5>

URI of a BAM file to access.  <i>e.g.</i>

{%highlight javascript%}
{name: "MyBAMTrack",
 bamURI: "http://server/myfile.sort.bam"}
{%endhighlight%}

<h5>baiURI</h5>

Explicitly specify the URI for the BAM index (.bam.bai) file.  If not specified, Dalliance tries appending '.bai' to the value
of the <code>bamURI</code> parameter.

<h3>Configuring a Tabix source</h3>

Dalliance can access Tabix-indexed flatfiles.  Currently supported payloads are VCF and BED, but other
formats may be added in future.

{%highlight javascript%}
{name: "Variants",
 uri: "http://server/myfile.vcf.gz",
 tier_type: 'tabix',
 payload: 'vcf'}
{%endhighlight%}

<h3>Configuring a Memstore source</h3>

Dalliance can directly access small textual files (currently supported formats are VCF, BED, and WIG -- more may be
added in future).  Since these don't support random access, the whole file will be loaded when the track is initialized,
hence "memory store" (memstore).  Configuration is very similar to tabix

{%highlight javascript%}
{name: "Variants",
 uri: "http://server/myfile.vcf",
 tier_type: 'memstore',
 payload: 'vcf'}
{%endhighlight%}

<h3><a id='jbrowse-rest'>Configuring a JBrowse-REST source</a></h3>

{%highlight javascript%}
{name: 'JBTest',
 jbURI: 'http://my.server.com/jbrowse/rest',
 jbQuery: 'type=IntergenicRegion'}
{%endhighlight%}

<h3><a id='ensembl-rest'>Configuring an Ensembl-REST source</a></h3>

Since Dalliance 0.9, data can be fetched from the <a href='http://beta.rest.ensembl.org/'>Ensembl-REST API</a>.

<h5>tier_type: ensembl</h5>

Required for Ensembl-REST source configurations

<h5>uri</h5>

Base URI for the server.  Unusually, this is <b>not</b> required for Ensembl-REST
sources, and defaults to <code>http://beta.rest.ensembl.org</code>.

<h5>species (currently required)</h5>

A species name as recognized by the Ensembl-REST API.  Future versions may
try to work this out automatically from the browser's <code>coordSystem</code>
parameter.

<h5>type</h5>

An array of feature types to fetch.

{%highlight javascript%}
{name:                 'e! transcripts',
 uri:                  'http://beta.rest.ensembl.org',
 tier_type:            'ensembl',
 species:              'human',
 type:                 ['transcript', 'exon', 'cds']}
{%endhighlight%}
